node/trieIterators/delegatedListProcessor.go

```diff
@@ -0,0 +1,212 @@
+package trieIterators
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"math/big"
+	"sort"
+	"strings"
+
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/core/vmcommon"
+	"github.com/ElrondNetwork/elrond-go/data"
+	"github.com/ElrondNetwork/elrond-go/data/api"
+	"github.com/ElrondNetwork/elrond-go/epochStart"
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/ElrondNetwork/elrond-go/vm"
+)
+
+type delegatedListProc struct {
+	*commonStakingProcessor
+	publicKeyConverter core.PubkeyConverter
+}
+
+// NewDelegatedListProcessor will create a new instance of delegatedListProc
+func NewDelegatedListProcessor(
+	accounts *AccountsWrapper,
+	blockChain data.ChainHandler,
+	queryService process.SCQueryService,
+	publicKeyConverter core.PubkeyConverter,
+) (*delegatedListProc, error) {
+	if accounts == nil || check.IfNil(accounts) {
```

Comment:
isn't `accounts == nil` redundant?

Reply:
no, it will panic if I give a nil accounts and have only the check.IfNil(accounts). Why? Because the compiler will translate 
accounts.IsInterfaceNil() call to accounts.AccountsHandler.IsInterfaceNil()